CONTRIBUTING.md

```diff
@@ -0,0 +1,102 @@
+2. Change the debug configuration to "WebJobs.Script.WebHost.Core"
+3. Add `AzureWebJobsScriptRoot` setting pointing at your test project
+   - You can add this variable a few ways:
+        1. Add a Environment Variable via a global variable (have to restart VS afterwards)
+        2. Add a setting to appsettings.json (be careful not to check it in)
+        3. Add an environment variable via the Debug configuration for the project (be careful not to check it in)
+   - You can create a simple hello world function app via the function core tools CLI. In a sample directory just run:
+      ```
+      func init
+      func new -l javascript -t httptrigger -n hello
+      echo $PWD
+      ```
+      The output of $PWD is your current directory, use that full path
+4. Click debug - this should launch a new terminal and browser window. If you created a new http triggered function named "hello", you should be able to add "api/hello" to your base URL and see your Function's response and the `context.log` statement in the terminal.
+
+If you want to test anything other than HTTP, it will require the `AzureWebJobsStorage` and `AzureWebJobsDashboard` settings get set to an Azure Storage Account connection string. You'll also need to add settings for any non-storage services you might want to connect to. You can do this via the same 3 methods described above.
+
+
+## Running tests (Visual Studio)
+There are three test projects in the WebJobs.Script solution
+ - WebJobs.Script.Scaling.Tests 
+ - WebJobs.Script.Tests
+ - WebJobs.Script.Test.Integration
+
+The only thing you need to set up for the tests is your Storage Account for the integration tests. You need to set the `AzureWebJobsStorage` and `AzureWebJobsDashboard` settings for WebJobs.Script.Tests.Integration project. The appsettings.json method is pretty clean, but you need to create it. Be sure "CopyToOutput" is true and rebuild afterwards for the appsettings.json file. To run end to end (E2E) tests, see the set up requirements for Samples under [Home](https://github.com/Azure/azure-webjobs-sdk-script/wiki).
+
+Then open up for your test explorer (CTRL+E, T) and click Run All. If any fail, you might have set something up wrong locally.
+
+## Change flow
+
+The general flow for making a change to the script host is:
+1. 🍴 Fork the repo (add the fork via `git remote add me <clone url here>`
+2. 🌳 Create a branch for your change (generally use dev) (`git checkout -b my-change`)
+3. 🛠 Make your change
+4. ✔️ Test your changes
+5. ⬆️ Push your changes to your fork (`git push me my-change`)
+6. 💌 Open a PR to the dev branch
+7. 📢 Address feedback and make sure tests pass (yes even if it's an "unrelated" test failure)
+8. 📦 [Rebase](https://git-scm.com/docs/git-rebase) your changes into a meaningful commits (`git rebase -i HEAD~N` where `N` is commits you want to squash)
+9. :shipit: Rebase and merge (This will be done for you if you don't have contributor access)
```

Comment:
As part of our normal flow, we squash/cleanup the commit history. Usually down to a single commit, more if the history is important

Reply:
That's the 8 step, but I call it "rebase". I should rename that to squash and mention it normally goes down to 1.